src/App.tsx

```diff
@@ -17,18 +43,35 @@
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                currentStatus={filterStatus}
+                onFilterChange={setFilterStatus}
+                query={query}
+                onQueryChange={setQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {!isLoading && todos.length > 0 ? (
+                <TodoList
+                  todos={getFilteredTodos}
+                  selectedTodo={isSelectedTodo}
+                  setSelectedTodo={setIsSelectedTodo}
+                />
+              ) : (
+                <Loader />
```

Comment:
The `Loader` component is displayed when `isLoading` is true or when there are no todos. If the intention is to show the loader only when loading, consider changing the condition to `isLoading ? <Loader /> : <TodoList ... />` to ensure the loader is only shown during loading.